pkg/apis/core/validation/validation.go

```diff
@@ -3008,12 +3008,19 @@ func ValidateNodeSelectorRequirement(rq core.NodeSelectorRequirement, fldPath *f
 func ValidateNodeSelectorTerm(term core.NodeSelectorTerm, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(term.MatchExpressions) == 0 {
-		return append(allErrs, field.Required(fldPath.Child("matchExpressions"), "must have at least one node selector requirement"))
+	if len(term.MatchExpressions) == 0 && len(term.MatchFields) == 0 {
+		return append(allErrs, field.Required(fldPath.Child("matchExpressions/matchFields"),
```

Comment:
this isn't a valid field path... what we typically do in situations like this is put the required error on the parent struct with a message. for example:

```
	if numVolumes == 0 {
		allErrs = append(allErrs, field.Required(fldPath, "must specify a volume type"))
	}
```